.github/workflows/R-CMD-check.yaml

```diff
@@ -85,6 +85,7 @@ jobs:
         run: |
           options(warn = 2)
           pak::local_install_dev_deps("cpp11test", dependencies = TRUE)
+          install.packages(".", repos = NULL, type = "source")
```

Comment:
cc @kevinushey @DavisVaughan just a FYI, because this has driven me insane over the past few days. 

Because this is a non-standard setup that involves an extra package (`cpp11test`) that lives under the source tree, we actually do need to install `cpp11` from locally, which is not done for usual testing (it just does `R CMD build + check`). 

So I ended up testing against cran version instead of local, so: 
 - none of the changes in header files were stressed, causing failures on ci. 
 - but locally everything was fine

And then because I'm always more suspicious of my local setup than the ci, I tried a bunch of things (see #318 for some drama ci-based testing)